Add tests for custom properties that use CSS-wide keywords

kizu opened the pull request:
After encountering several inconsistencies in how custom properties use CSS-wide keywords, I did an extensive testing for most of the cases I could think of ([CodePen](https://codepen.io/kizu/pen/rNgpXwL?editors=1100)), and decided to add these as tests.

When I am testing it, Chrome succeeds in all cases, Firefox fails in 5 of them, and Safari TP in 8 of them.

Relevant specs:

- [CSS Custom Properties for Cascading Variables Module Level 1](https://drafts.csswg.org/css-variables/#ref-for-css-wide-keywords%E2%91%A0), specifically this part (among some others):

    > The CSS-wide keywords can be used in custom properties, with the same meaning as in any another property.

- [CSS Values and Units Module Level 3](https://www.w3.org/TR/css-values-3/#common-keywords) defining `initial`, `inherit` and `unset`
- [CSS Cascading and Inheritance Level 4](https://drafts.csswg.org/css-cascade/#valdef-all-revert) defining `revert`
- [CSS Cascading and Inheritance Level 5](https://drafts.csswg.org/css-cascade-5/#revert-layer) defining `revert-layer`

There were a couple of disjointed tests for two of the assertions that I found ([one](https://wpt.fyi/results/css/css-variables/wide-keyword-fallback-002.html?label=master&label=experimental&aligned&q=wide-keyword), [two)](https://wpt.fyi/results/css/css-variables/revert-layer-in-fallback.html?label=master&label=experimental&aligned&q=revert-layer), but I decided that it won't hurt to have those duplicate in one test file for all the CSS-wide keywords.

I also tested many various permutations, including some obvious ones, and this was good, as some of the failures I found were not obvious, like Firefox failing with two cases for registered properties with `initial` and `inherit` values.

I did find only one bug related to the failures I found: https://bugs.webkit.org/show_bug.cgi?id=271136 — there might be others, but I am planning to fill separate bugs for Firefox and Safari anyway. Here they are:

- https://bugzilla.mozilla.org/show_bug.cgi?id=1903397
- https://bugs.webkit.org/show_bug.cgi?id=275629

kizu committed:
Add tests for custom properties that use CSS-wide keywords

svgeesus commented on css/css-variables/variable-css-wide-keywords.html:
```diff
@@ -0,0 +1,330 @@
+<!DOCTYPE html>
+<head>
+<title>CSS Custom Properties: Using CSS-wide keywords</title>
+<link rel="help" href="https://drafts.csswg.org/css-variables/#ref-for-css-wide-keywords%E2%91%A0">
```
I worry a little about the ① which is how Bikeshed de-dupes ids.

tabatkins replied:
Yes, please don't use those `#ref-for` IDs, they're auto-generated and not remotely stable, and created just so the index can link to those elements. It would be more reasonable to use a [text fragment](https://developer.mozilla.org/en-US/docs/Web/Text_fragments); then at least it wouldn't break when someone added or removed a reference above that point in the spec.

But really, if you need an anchor to that paragraph, we can just add an id to the paragraph. Please do that instead.

kizu replied:
@tabatkins I'll be happy to change this. I did not quickly find a way to add an `id` to a paragraph in bikeshed, what could be the best way to do this in this case?

Here is this place in the raw spec: https://github.com/w3c/csswg-drafts/blob/main/css-variables-1/Overview.bs#L219

I wonder if there is a way to make the ref link in this case to be stable instead (provide a hard-coded id to it?)

Or maybe a more simple fix in this case would be for me to just link to the section containing this paragraph, in this case, https://drafts.csswg.org/css-variables/#defining-variables ?

dbaron replied:
Maybe the simple answer is:
* link to the section
* quote the spec sentence in a `<meta name=assert>` [as documented](https://web-platform-tests.org/writing-tests/css-metadata.html#test-assertions)

dbaron replied:
That said, if you want to add an `id` to the paragraph in the spec I think you can just start it with `<p id="my-id">`.

tabatkins replied:
Yeah, I propose linking to a section. But if you really do want to link to a paragraph, the answer is "add a `<p>` tag" - just because we use Markdown paragraphs for nearly everything doesn't mean HTML paragraphs aren't allowed. ^_^

kizu replied:
Done: changed the link to point to the section, and added a `<meta name="assert">`

svgeesus reviewed (approved):
Looks nice overall

kizu committed:
Update variable-css-wide-keywords.html

Fix the link to the spec (link to the section), and add a meta assert with the quoted paragraph in the spec.
